src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -60,12 +60,84 @@ struct WeightedEdge
   }
 };
 
-template <class BVH, class Labels, class Edges, class Metric, class Radii,
-          class LowerBounds>
+class DirectedEdge
+{
+public:
+  unsigned long long directed_edge = ULLONG_MAX;
```

Comment:
Does this need to be exposed given that we have `source()`, `target()`, and `reverse()`?

Reply:
Yes we do atomic min on `directed_edge`